### DISCUSSION TITLE
Unable to add Azure Table Storage

### DISCUSSION THREAD

**Discussion by githubacct54385** [CompositionalIT/farmer#589] [Q&A] [answered]
Hello,
     I'm trying to follow the instructions to add a storage account with azure tables on this page in the documentation: [Storage Account](https://compositionalit.github.io/farmer/api-overview/resources/storage-account/)

I created a file `Program.fs` with the following:
```fs
open Farmer
open Farmer.Builders

let storage = storageAccount {
    name "btdevstracct"
    add_table "mytable" // this causes a build error
}

let deployment = arm {
    location Location.NorthEurope
    add_resource storage
}

deployment
|> Writer.quickWrite "myFirstTemplate"
```

My fsproj file is below:
```
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Program.fs" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Update="FSharp.Core" Version="4.*" />
    <PackageReference Include="Farmer" Version="1.*" />
  </ItemGroup>
</Project>
```

Build the project
`dotnet build`

Error as follows:
`D:\AzureRepos\path\to\project\Program.fs(5,5): error FS0708: This control construct may only be used if the computation expression builder defines a 'For' method [D:\path\to\the\project\project.fsproj]
D:\path\to\the\project\Program.fs(11,5): error FS0041: A unique overload for method 'AddResource' could not be determined based on type information prior to this program point. A type annotation may be needed.↔↔Known types of arguments: ArmConfig * 'a↔↔Candidates:↔ - member ArmBuilder.AddResource : state:ArmConfig * input:Builder -> ArmConfig↔ - member ArmBuilder.AddResource : state:ArmConfig * input:IArmResource -> ArmConfig↔ - member ArmBuilder.AddResource : state:ArmConfig * input:IBuilder -> ArmConfig [D:\path\to\the\project\project.fsproj]`

Am I missing something about an arm config?

**Comment by isaacabraham** [comment, marked as answer]
Hi @githubacct54385 . This is only available in 1.5 which is still in beta. It's available on NuGet, but I suspect you've pulled down 1.4 which won't have it yet.

**Reply by isaacabraham** [reply]
Apologies for the totally unreadable error message - this is one of the "costs" of the JSON-style syntactic sugar that Farmer uses. cc @cartermp.

**Reply by githubacct54385** [reply]
Hello @isaacabraham yes this did the trick.  I am now using the following .fsproj file.  Thank you for your help!

```fsproj
<?xml version="1.0" encoding="utf-8"?>
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Program.fs" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Update="FSharp.Core" Version="5.0.1" />
    <PackageReference Include="Farmer" Version="1.5.0-beta2" />
  </ItemGroup>
</Project>
```
